Automattic/sensei
# Merge answer feedback feature branch

Fixes #4336

Fixes #1484
Fixes #4312
Fixes #3288

### Changes proposed in this Pull Request

* Add gradual controls for answer feedback to the quiz
* Introduce block-based answer feedback, and allow separate correct an incorrect answer feedback

### Testing instructions

* Create questions with answer feedback in the `master` branch. Finish a quiz with a learner and grade it.
* Switch to `feature/answer-feedback`. Open the quiz frontend with the learner and make sure everything still works, the previous answer feedback text is displayed.
* Edit the quiz. Check that the previous answer feedback text is converted to blocks for the questions. 
* Change feedback content, save & check the quiz feedback with the learner again.
* Open the grading screen and check that answer feedback is displayed correctly there


### New/Updated Hooks

* See #4408


### Deprecated Code

* See #4408
* `_answer_feedback` meta on questions is replaced with blocks in the question post content.


### Screenshot / Video

#### Frontend

<img width="749" alt="image" src="https://user-images.githubusercontent.com/176949/142876859-8ca4bc61-2c88-40ff-afe2-ea3c7e82b950.png">

#### Quiz settings

<img width="261" alt="image" src="https://user-images.githubusercontent.com/176949/142876920-fdce5d39-9859-4883-9fbd-00d978b41e38.png">

#### Question block

<img width="709" alt="image" src="https://user-images.githubusercontent.com/176949/142877001-b82df061-c1ea-4479-9142-bd5b684c49a8.png">

#### Grading

<img width="1034" alt="image" src="https://user-images.githubusercontent.com/176949/142877063-27643796-5e00-499f-ba82-574ef046c66f.png">
